chef/automate
# scan jobs project filtering

### :nut_and_bolt: Description: What code changed, and why?

Removing bits of logic and adding bits of logic to enable project fitlering on scan job results.

### :+1: Definition of Done
A user can specify an environment for a node (via tags), create a project that has a rule for that environment value, and the report lands in the appropriate project.

### :athletic_shoe: How to Build and Test the Change
- go_update_component compliance-service
- create a project with a rule, environment == "test-env"
- run `load scan jobs` OR create a credential, add a node with an environment tag matching the value set for the project, then add a scanjob for that node
- check that the reports are correctly associated with the project (the two ec2 nodes should end up there)

~**test not working yet**~ done
~**still need to add docs**~ done
~**make sure the project update path works now that report read has a job id on it**~ done

### :white_check_mark: Checklist
